pkg/kds/context/context_test.go

```diff
@@ -433,4 +439,87 @@ var _ = Describe("Context", func() {
 			)
 		})
 	})
+	Describe("GlobalResourceMapper", func() {
+		type config struct {
+			storeType          config_store.StoreType
+			k8sSystemNamespace string
+		}
+
+		type testCase struct {
+			config                     config
+			name                       string
+			expectedName               string
+			isResourcePluginOriginated bool
+		}
+
+		genConfig := func(caseCfg config) kuma_cp.Config {
+			cfg := kuma_cp.DefaultConfig()
+
+			if caseCfg.storeType != "" {
+				cfg.Store.Type = caseCfg.storeType
+			}
+
+			if caseCfg.k8sSystemNamespace != "" {
+				cfg.Store.Kubernetes.SystemNamespace = caseCfg.k8sSystemNamespace
+			}
+
+			return cfg
+		}
+
+		resource := func(given testCase) model.Resource {
+			descriptor := model.ResourceTypeDescriptor{
+				IsPluginOriginated: given.isResourcePluginOriginated,
+			}
+
+			meta := &test_model.ResourceMeta{Name: given.name}
+
+			return &test_model.Resource{TypeDescriptor: descriptor, Meta: meta}
+		}
+
+		DescribeTable("system namespace suffix from in resource names",
+			func(given testCase) {
+				// given
+				ctx := stdcontext.Background()
+				rm := manager.NewResourceManager(memory.NewStore())
+				cfg := genConfig(given.config)
+				kdsCtx := context.DefaultContext(ctx, rm, cfg)
+
+				// when
+				out, _ := kdsCtx.GlobalResourceMapper(resource(given))
+
+				// then
+				Expect(out.GetMeta().GetName()).To(Equal(given.expectedName))
+			},
+			Entry("should be removed when store type is kubernetes "+
+				"and resource is plugin originated", testCase{
+				isResourcePluginOriginated: true,
+				config: config{
+					storeType:          config_store.KubernetesStore,
+					k8sSystemNamespace: "custom-namespace",
+				},
+				name:         "foo.custom-namespace",
+				expectedName: "foo",
+			}),
+			Entry("shouldn't be removed when store type is kubernetes "+
+				"and resource isn't plugin originated", testCase{
+				isResourcePluginOriginated: false,
+				config: config{
+					storeType:          config_store.KubernetesStore,
+					k8sSystemNamespace: "custom-namespace",
+				},
+				name:         "foo.custom-namespace",
+				expectedName: "foo.custom-namespace",
+			}),
+			Entry("shouldn't be removed when store type is not kubernetes ",
+				testCase{
+					isResourcePluginOriginated: true,
+					config: config{
+						storeType:          config_store.MemoryStore,
```

Comment:
can we use postgres here without doing any setup?

Reply:
we can

Reply:
addressed